### PR TITLE
windows: add device request "kds_cu_info" to retrieve compute units data

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Added kds_cu_info device request in windows to get the KDS CU information.
Fixed missing type entry in ReportDynamicRegion code

Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

### CODE REVIEW

File: src/runtime_src/core/pcie/windows/device_windows.cpp [hunk 1]
```diff
@@ -884,6 +884,8 @@ struct rom
       return static_cast<query::rom_ddr_bank_count_max::result_type>(hdr.DDRChannelCount);
     case key_type::rom_fpga_name:
       return std::string(reinterpret_cast<const char*>(hdr.FPGAPartName));
+    case key_type::rom_time_since_epoch:
```
Comment (rajkumar-xilinx) [inline src/runtime_src/core/pcie/windows/device_windows.cpp:887]: rom_time_since_epoch is needed in the xbutil validate flow, so moved this entry here to available for userpf device node as well.

**Comment by rajkumar-xilinx** [comment]
Output looks like below after loading bandwidth xclbin.
```
>xbutil examine -r dynamic-regions

-----------------------------------------------
1/1 [0000:00:00.0] : xilinx_u250_xdma_201830_2
-----------------------------------------------
Xclbin UUID
  C5B9A584-9B70-4902-AE32-ADDF5C1C6E0C

Compute Units
  PL Compute Units
    Index   Name                          Base_Address    Usage   Status
    0       bandwidth1:kernel_1           0x1800000       0       (--)
    1       bandwidth2:kernel_2           0x1810000       0       (--)

  PS Compute Units
    Index   Name                          Base_Address    Usage   Status
```

**Comment by xilinxgitops** [comment]
retest this please

**Comment by xilinxgitops** [comment]
retest this please

**Review by arpitxilinx — APPROVED** [review]

File: src/runtime_src/core/tools/common/ReportDynamicRegion.cpp [hunk 2]
```diff
@@ -131,6 +131,7 @@ populate_cus(const xrt_core::device *device)
         ptCu.put( "name",			layout->m_ip_data[i].m_name);
         ptCu.put( "base_address",		boost::str(boost::format("0x%x") % base_addr));
         ptCu.put( "usage",			usage);
+        ptCu.put( "type",			enum_to_str(cu_type::PL));
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/tools/common/ReportDynamicRegion.cpp:134]: What is the type representing here?

Comment (rajkumar-xilinx) [reply src/runtime_src/core/tools/common/ReportDynamicRegion.cpp:134]: kernel type PL or PS. populate_cus_new() had this change but this function populate_cus doesn't have it. So, added it here also. On windows XRT, populate_cus() function is getting invoked, and no node called "type" error is coming without this change.

**Review by chvamshi-xilinx — COMMENTED** [review]
Looks good. I have one query about "type"

File: src/runtime_src/core/tools/common/ReportDynamicRegion.cpp (hunk 2 above)
Comment (chvamshi-xilinx) [reply src/runtime_src/core/tools/common/ReportDynamicRegion.cpp:134]: why cant we use same function in both linux and windows?  populate_cus_new should be called here as well right?

Comment (rajkumar-xilinx) [reply src/runtime_src/core/tools/common/ReportDynamicRegion.cpp:134]: kds_mode is 0 on windows, hence populate_cus() is being invoked.

Comment (chvamshi-xilinx) [reply src/runtime_src/core/tools/common/ReportDynamicRegion.cpp:134]: Got it. Thanks Raj for the clarification.

**Review by chvamshi-xilinx — APPROVED** [review]

**Review by stsoe — APPROVED** [review]

**Pull request merged**